src/app/dashboard/Missions/MissionEditView.jsx

```diff
@@ -0,0 +1,368 @@
+    </Container>
+  );
+};
+const MissionTypeRow = ({ classes, mission }) => {
+  let missionType;
+  switch (mission?.type) {
+    case "foodbox":
+    default:
+      missionType = "Food Box";
+  }
+  return (
+    <Box marginTop="32px">
+      <H3>{missionType}</H3>
+    </Box>
+  );
+};
+const MissionStatusRow = ({ classes, mission }) => {
+  let status = mission?.status;
+  let missionStatusText;
+  switch (status) {
+    case Mission.Status.unassigned:
+      missionStatusText = "Looking for volunteer";
+      break;
+    default:
+      missionStatusText = status;
+      break;
+  }
+  return (
+    <Row Icon={PanToolIcon} classes={classes}>
+      {missionStatusText}
+    </Row>
+  );
+};
+
+const MissionFundedStatusRow = ({ classes, mission }) => {
+  let missionFundedStatusText;
+  switch (mission?.fundedStatus) {
+    case Mission.FundedStatus.fundedbydonation:
+      missionFundedStatusText = "Funded By Donation";
+      break;
+    case Mission.FundedStatus.fundedbyrecipient:
+      missionFundedStatusText = "Funded By Recipient";
+      break;
+    case Mission.FundedStatus.notfunded:
+    default:
+      missionFundedStatusText = "Not Yet Funded";
+  }
+  return (
+    <Row Icon={AttachMoneyIcon} classes={classes}>
+      {missionFundedStatusText}
+    </Row>
+  );
+};
+
+const FoodBoxDetailsRow = ({ classes, details }) => {
+  return (
+    <Box>
+      {details?.needs?.map((box, index) => (
+        <Grid key={index} container className={classes.foodBoxDetailContainer}>
+          <Grid className={classes.foodBoxDetailQuantity}>
+            <b>{box?.quantity}</b>
+          </Grid>
+          <Grid className={classes.foodBoxDetailName}>
+            <b>{box?.name}</b>
+          </Grid>
+        </Grid>
+      ))}
+    </Box>
+  );
+};
+
+const MissionDetailsRow = ({ classes, mission }) => {
+  let type = mission?.type;
+  let details = mission?.missionDetails;
+  if (type === "foodbox") {
+    return <FoodBoxDetailsRow details={details} classes={classes} />;
+  }
+  return null;
+};
+
+/**
+ * Component for editing mission details
+ * @component
+ */
+const MissionEditView = ({ mission, toListView }) => {
+  const classes = useStyles();
+
+  const { handleChange, values } = useForm(mission);
+
+  const pickUpLocationLabel = mission?.pickUpLocation?.label;
+  const recipientDisplayName = _.get(mission, "recipientDisplayName");
+  const recipientPhoneNumber = _.get(mission, "recipientPhoneNumber");
+  const deliveryNotes = _.get(mission, "deliveryNotes");
+
+  const props = { classes, mission };
+
+  function changeFormValue(name, value) {
+    handleChange({ target: { name, value } });
+  }
+
+  function handleChangeLocation(data) {
+    const { location } = data;
+    changeFormValue("location", location);
+  }
+
+  function handleChangePickUpLocationLabel(e) {
+    e.preventDefault();
+    changeFormValue("pickUpLocationLabel", e.target.value);
+  }
+
+  function handleChangeRecipientPhoneNumber(e) {
```

Comment:
fyi you only need to make these functions for non standard inputs.

So normally you can just do
```js
<Input
name ='phone'
onChange={handleChange}
>
{values.phone}
</Input>
```

Reply:
Ok, extraneous functions removed.